src/ui/segments/data-table/elements/use-data-table-columns.tsx

```
@@ -239,12 +239,19 @@ export function useDataTableColumns<T>({
   );
 
   if (dataType) {
-    return columns.sort((a, b) =>
-      a.key && b.key
-        ? ViewsDefinitionRegistry[dataType].columns.indexOf(a.key as EntityCoreFields) -
-          ViewsDefinitionRegistry[dataType].columns.indexOf(b.key as EntityCoreFields)
-        : -1
-    );
+    return columns.sort((a, b) => {
+      const defs = ViewsDefinitionRegistry[dataType];
+      if (!defs) {
+        throw new Error(
+          `Cannot find any item in ViewsDefinitionRegistry for dataType "${dataType}"!`
```

if this error is not handled in the upper scope, then may be the message should be more user friendly, you can add technical details (as mentioning `ViewsDefinitionRegistry`) in the cause property as
`new Error("message", { cause: any})`,